packages/connect-wallet-modal/src/connectButtons/ConnectWC.tsx

```diff
@@ -97,30 +98,40 @@ export const ConnectWC: FC<ConnectWCProps> = (props: ConnectWCProps) => {
         redirectionWindow?.close();
       }
     } else {
-      await connectAsync({ connector });
-      onSuccess();
+      setLoadingWalletId(walletId);
+      await connectAsync(
+        { connector },
+        {
+          onSettled: () => {
```

Comment:
Can you check if we need those callbacks? Perhaps you can use try,catch,finally.

Reply:
Why? These are documented callbacks, they work, why should I change them?

Reply:
Because it makes code more complicated. If we can use try/catch/finally it will much easier to understand, e.g. right now we need to refer to docs for onSettled.